File: fortran/fortran90/Dart/Fortran90LexerBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:convert';

abstract class Fortran90LexerBase extends Lexer
{
    Fortran90LexerBase(CharStream input) : super(input)
    {
    }

    bool IsColumnZero()
    {
        return charPositionInLine == 0;
    }

    bool VerifyNotOperator()
    {
        var c1 = this.inputStream.LA(1);
        if (c1 == "a".codeUnitAt(0))
        {
            var c2 = this.inputStream.LA(2);
            if (c2 == "n".codeUnitAt(0))
            {
                var c3 = this.inputStream.LA(3);
                if (c3 == "d".codeUnitAt(0))
                {
                    var c4 = this.inputStream.LA(4);
                    if (c4 == ".".codeUnitAt(0))
                    {
                        return false;
                    }
                }
            }
        }
        else if (c1 == 'o')
        {
            var c2 = this.inputStream.LA(2);
            if (c2 == "r".codeUnitAt(0))
            {
                var c3 = this.inputStream.LA(3);
                if (c3 == ".".codeUnitAt(0))
                {
                    return false;
                }
            }
        }
        return true;
    }
}

File: fortran/fortran90/Java/Fortran90LexerBase.java
import org.antlr.v4.runtime.*;

public abstract class Fortran90LexerBase extends Lexer
{
    public Fortran90LexerBase(CharStream input)
    {
        super(input);
    }

    protected boolean IsColumnZero()
    {
    return this.getCharPositionInLine() == 0;
    }

    public boolean VerifyNotOperator()
    {
        var c1 = this.getInputStream().LA(1);
        if (c1 == 'a')
        {
            var c2 = this.getInputStream().LA(2);
            if (c2 == 'n')
            {
                var c3 = this.getInputStream().LA(3);
                if (c3 == 'd')
                {
                    var c4 = this.getInputStream().LA(4);
                    if (c4 == '.')
                    {
                        return false;
                    }
                }
            }
        }
        else if (c1 == 'o')
        {
            var c2 = this.getInputStream().LA(2);
            if (c2 == 'r')
            {
                var c3 = this.getInputStream().LA(3);
                if (c3 == '.')
                {
                    return false;
                }
            }
        }
        return true;
    }
}
